lib/service/connect.go

```diff
@@ -160,6 +161,15 @@ func (process *TeleportProcess) connect(role types.SystemRole) (conn *Connector,
 		process.log.Infof("Connecting to the cluster %v with TLS client certificate.", identity.ClusterName)
 		clt, err := process.newClient(process.Config.AuthServers, identity)
 		if err != nil {
+			// In the event that a user is attempting to connect a machine to
+			// a different cluster it will give a cryptic warning about an
+			// unknown certificate authority. Unfortunately we cannot intercept
+			// this error as it comes from the http package before a request is
+			// made. So provide a more user friendly error as a hint of what
+			// they can do to resolve the issue.
+			if strings.Contains(err.Error(), "certificate signed by unknown authority") {
+				return nil, trace.NewAggregate(errors.New("Was this node already registered to a different cluster? To re-join the node, remove `/var/lib/teleport` and try again"), err)
```

Comment:
Rather than aggregating with a new error (which itself is an aggregate of 3 other errors), should we just log the "Was this node already registered.." message here and return the original error unmodified?

Reply:
Yeah that's a good idea, I've updated it to do just that.